shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -528,17 +528,29 @@ public boolean onTouchEvent(MotionEvent event) {
         packet.order(ByteOrder.LITTLE_ENDIAN);
 
         int maskedAction = event.getActionMasked();
-        // ACTION_UP, ACTION_POINTER_UP, ACTION_DOWN, and ACTION_POINTER_DOWN
-        // only apply to a single pointer, other events apply to all pointers.
-        if (maskedAction == MotionEvent.ACTION_UP || maskedAction == MotionEvent.ACTION_POINTER_UP
-                || maskedAction == MotionEvent.ACTION_DOWN || maskedAction == MotionEvent.ACTION_POINTER_DOWN) {
-            addPointerForIndex(event, event.getActionIndex(), packet);
+        int pointerChange = getPointerChangeForAction(event.getActionMasked());
+        if (maskedAction == MotionEvent.ACTION_DOWN || maskedAction == MotionEvent.ACTION_POINTER_DOWN) {
+            // ACTION_DOWN and ACTION_POINTER_DOWN always apply to a single pointer only.
+            addPointerForIndex(event, event.getActionIndex(), pointerChange, packet);
+        } else if (maskedAction == MotionEvent.ACTION_UP || maskedAction == MotionEvent.ACTION_POINTER_UP) {
+            // ACTION_UP and ACTION_POINTER_UP may contain position updates for other pointers.
+            // We are sending a special update event instead of a move event here, in order to
+            // help the framework reassemble the original Android event later, should it need
+            // to forward it to a PlatformView.
```

Comment:
nit: I don't think the reason to do it here is just to help the framework reassemble the original Android event, we do it to make sure we don't lose any data. I'd consider rephrasing the comment, maybe mention the PlatformView as one case that is using this information but I would feel more comfortable if it didn't sound like the sole purpose of sending the data here is for the PlatformView.